pages/docs/couple-your-code/couple-your-code-just-in-time-mapping.md

```diff
@@ -0,0 +1,134 @@
+
+1. `api-access` needs to be enabled on the `receive-mesh` (similar to the direct access), "ReceivedMeshName" in our example above.
+2. The `read-data` and `write-data` tags need to use the received mesh with api-access enabled.
+3. The mapping only defines (depending on the direction) a "from" or a "to" mesh. For read-consistent mappings, only the "from" mesh must be defined. For write-conservative mappings, only the "to" mesh must be defined. In both cases, the defined mesh needs to match the received mesh with api-access enabled (which is also used in the `read-data` or `write-data` tag).
+
+At its core, the API now makes use of two new API functions, called `mapAndReadData` and `mapAndWriteData`. Both functions are very similar to the analogous functions `readData` and `writeData`: the only difference is that the functions take spatial coordinates as function arguments instead of static vertexIDs stemming from the preCICE initialization. In addition to `mapAndReadData` and `mapAndWriteData`, the access region needs to be defined by "SolverOne" using `setMeshAccessRegion`. The full code example reads the following:
+
+```cpp
+    // Note that "ReceivedMeshName" is defined and received
+    // from another participant
+    const std::string otherMesh = "ReceivedMeshName";
+
+    // Allocate and fill a 'boundingBox' to define an access region
+    // in our example, we use the unit cube.
+    // Assuming dim == 3, means that the bounding box has
+    // dim * 2 == 6 elements.
+    std::vector<double> boundingBox {
+        0, 0, 0, // minimum corner
+        1, 1, 1 // maximum corner
+    };
+
+    // Define region of interest, where we want to obtain the direct access.
+    // See also the API documentation of this function for further notes.
+    precice.setMeshAccessRegion(otherMesh, boundingBox);
+
+    // initialize preCICE as usual
+    // initializing data (requiresInitialData) before calling initialize is not possible, as we first need to exchange the
+    // meshes before being able to access the mesh
+    precice.initialize();
+
+    // until this point, everything was just the same as for
+    // the direct mesh access, now we enter the main time loop
+    // and use dedicated read and write functions
+    while (precice.isCouplingOngoing()) {
+
+        // reading data requires a time stamp (relative read
+        // time) for time interpolation, which we need below
+        // we simply select here the end of the time window
+        double dt = couplingInterface.getMaxTimeStepSize();
+
+        // The just-in-time reading function: its interface
+        // is very similar to the conventional readData
+        // function, only the vertexIDs are now dynamic
+        // coordinates:
+
+        // we need a vector, where we store the data we read
+        std::vector<double> velocityValue(1);
```

Comment:
If we use velocities and forces, we should go to vectorial data. Will otherwise be confusing.

Reply:
used vectorial data